docs/mgmt/readme.md

```diff
@@ -625,6 +626,42 @@ list-exception:
 - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/availabilitySets/{availabilitySetName}
 ```
 
+### Partial resources
+
+Some operations might be operations of a resource from another RP, or some resources might have a parent resource from another RP. 
+
+For instance, in the RP `compute`, we have a resource `VirtualMachine` with the path of `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}`, in the meantime, in the RP `network`, which is completely a different SDK, we have an operation with the path of `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/publicIps` which is clearly listing something on the virtual machine. During the generation of network RP, the generator has no knowledge of the resources in another RP, therefore by default this operation will be generated like the following code as an operation on the `ResourceGroupResource` because this is the best parent the generator could find within this RP.
+```csharp
+public static partial class NetworkExtensions
+{
+    public static Pageable<PublicIPAddress> GetPublicIPs(this ResourceGroupResource resourceGroup, string vmName)
+    {
+        /* the implementation */
+    }
+} 
+```
+
+The following configuration
+```yaml
+partial-resources:
+  /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}: VirtualMachine
+```
+will tell the generator that this path `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}` is also a valid resource, and the generator will put it into the list of its known resources, and generates a "partial resource" from it and the operation in above example will show up here:
+```csharp
+public partial class VirtualMachineNetworkResource
+{
+    public virtual Pageable<PublicIPAddress> GetPublicIPs()
+    {
+        /* the implementation */
+    }
+}
+```
+
+The naming pattern of a partial resource is like `[ResourceName][RPName]Resource`, where `ResourceName` is the value you assigned in the configuration, `RPName` is the name of this RP (for instance, `Network`). The name of a partial resource is also configurable using the `request-path-to-resource-name` configuration, please see [change resource name](#change-resource-name) section for more details.
```

Comment:
Not sure, but is there case that the `ResourceName` = `RPName` ? Or similar, then the name of partial resource will be weird, so how about have a conjunction between them?

Reply:
Considering that the occurrence of partial resource will be not quite common, I believe it would be even more rarely that your edge case could happen. Therefore I did not introduce any "automatic logic" when getting a name for this partial resource.
We could always use `request-path-to-resource-name` to customize the name of a partial resource.
Here we have an example in the test project: https://github.com/Azure/autorest.csharp/blob/55ca9538793cfc0691aecd0b01e1dfd713ad9593/test/TestProjects/MgmtPartialResource/readme.md

Reply:
Ok, this is LGTM